crypto/evp/pmeth_lib.c

```diff
@@ -890,7 +890,8 @@ static int legacy_ctrl_to_param(EVP_PKEY_CTX *ctx, int keytype, int optype,
         case EVP_PKEY_CTRL_CMS_DECRYPT:
         case EVP_PKEY_CTRL_CMS_ENCRYPT:
 # endif
-            if (ctx->pmeth->pkey_id != EVP_PKEY_RSA_PSS)
+            /* TODO (3.0) Temporary hack, this should probe */
+            if (!EVP_PKEY_is_a(EVP_PKEY_CTX_get0_pkey(ctx), "RSASSA-PSS"))
```

Comment:
Can get0_pkey return NULL? Will EVP_PKEY_is_a handle ti gracefully?

Do we care about a far fetched case in a temporary solution?

Reply:
> Do we care about a far fetched case in a temporary solution?

I've seen crashes because `ctx->pmeth == NULL`

Reply:
> Can get0_pkey return NULL? Will EVP_PKEY_is_a handle ti gracefully?

Too answer that, you need to look here:

https://github.com/openssl/openssl/blob/fbc6efbbc16fc1410f776431395ea2b2ad691d4c/crypto/evp/pmeth_lib.c#L913-L920

See how it checks that an operation is ongoing?  The basic assumption, as far as I know, is that you can't have an initialized operation without a key.